src/app/views/users/UsersList.jsx

```diff
@@ -0,0 +1,174 @@
+                      isDismissable: true,
+                  };
+                  notifications.add(notification);
+                  break;
+              }
+              case "UNEXPECTED_ERR": {
+                  const notification = {
+                      type: "warning",
+                      message:
+                        "An unexpected error's occurred whilst trying to get users. You may only be able to see previously loaded information.",
+                      isDismissable: true,
+                  };
+                  notifications.add(notification);
+                  break;
+              }
+              case "FETCH_ERR": {
+                  const notification = {
+                      type: "warning",
+                      message:
+                        "There's been a network error whilst trying to get users. You may only be able to see previously loaded information.",
+                      isDismissable: true,
+                  };
+                  notifications.add(notification);
+                  break;
+              }
+              default: {
+                  log.event("Unhandled error fetching users", log.data({ status_code: error.status }), log.error(error));
+                  const notification = {
+                      type: "warning",
+                      message:
+                        "An unexpected error's occurred whilst trying to get users. You may only be able to see previously loaded information.",
+                      isDismissable: true,
+                  };
+                  notifications.add(notification);
+                  break;
+              }
+          }
+          console.error("Error getting all users:\n", error);
+      });
+};
+
+export const mapUserToState = (rootPath, user) => {
+    try {
+        const id = user.email;
+        const title = `${user.forename} ${user.lastname}`;
+        const details = [user.email];
+        const url = `${rootPath}/users/${id}`;
+        return { ...user, id, title, details, url };
+    } catch (error) {
+        const notification = {
+            type: "warning",
+            message: "Error mapping users to state",
+            isDismissable: true,
+            autoDismiss: 3000,
+        };
+        notifications.add(notification);
+        console.error("Error mapping users to state: ", error);
+        log.event("Error mapping users to state", log.error(error));
+        return false;
+    }
+};
+
+const propTypes = {
+    rootPath: PropTypes.string.isRequired,
+    dispatch: PropTypes.func.isRequired,
+    children: PropTypes.element,
+    params: PropTypes.shape({
+        userID: PropTypes.string,
+    }).isRequired,
+    loggedInUser: PropTypes.object,
+    users: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+};
+
+export const UsersList = (props) => {
+    const [isFetchingUsers, setIsFetchingUsers] = useState(false);
+    const isAdmin = auth.isAdmin(props.loggedInUser);
```

Comment:
Haven't run the code, but just a double-check does this work if you sign in then in a new tab go straight to this screen? I remember there were issues around checking isAdmin on load and redirecting as a result. useEffect is called after first renderer. But, I don't know if the `auth` would have been updated in time as it makes a request to permissions or something and if not returned in time the user will be kicked out. I think if the user was always coming from the collections screen it should work but if first load on react side of Florence I am not 100% sure. Before the upgrade this was done in the component will mount lifecycle function, which was called before first render, but we don't have that available anymore.

Reply:
I have tried the steps you described. When I go straight to this screen in a new tab the page does load correctly.